extensions/amp-render/1.0/amp-render.js

```diff
@@ -132,7 +147,12 @@ export const getJsonFn = (element) => {
   if (isAmpScriptSrc(src)) {
     return (src) => getAmpScriptJson(element.getAmpDoc(), src);
   }
-  return () => batchFetchJsonFor(element.getAmpDoc(), element);
+  return (_src, shouldRefresh = false) =>
+    batchFetchJsonFor(
+      element.getAmpDoc(),
+      element,
+      buildOptionsObject(element, shouldRefresh)
```

Comment:
Can this be saved to an internal variable instead of calculated each time?

Reply:
If so, we should probably cache the higher-level `getJsonFn()` result. We have similarly costly operations above like `getAmpStateJson()` etc.

But I don't know how frequently this is expected to be called. If the Preact layer calls this inside a hook, then we should leave the caching concern to its deps.

Reply:
So while `getJsonFn()` is not cached, its return value passed as `getJson` prop is called inside `useEffect` & `useCallback` hooks (https://github.com/ampproject/amphtml/blob/main/extensions/amp-render/1.0/component.js#L55,L76). Does that suffice?

Reply:
@dmanek Yep, if we know this is only called when needed then we probably don't need a cache.